src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -212,13 +212,28 @@ const PlrDaoStakingTransactionBlock = ({
 
   const hasEnoughPLR =
     totalKeyBasedPLRTokens >= MAX_PLR_TOKEN_LIMIT || totalSmartWalletPLRTokens >= MAX_PLR_TOKEN_LIMIT;
-  const enableAssetBridge =
-    selectedFromNetwork?.chainId !== CHAIN_ID.POLYGON &&
-    selectedFromAsset?.symbol === plrDaoAssetPerChainId[STAKING_CHAIN_ID].symbol;
-  const enableAssetSwap = selectedFromAsset?.symbol !== plrDaoAssetPerChainId[STAKING_CHAIN_ID].symbol;
+
+  const hasNotEnoughPLRForPolygon = useCallback(() => {
+    const selectedChainBalance = accounts.find((acc) => acc.chainId === CHAIN_ID.POLYGON);
+    if (!selectedChainBalance) return false;
+
+    if (selectedAccountType === AccountTypes.Contract) {
+      return selectedChainBalance['smartWallet'] < MAX_PLR_TOKEN_LIMIT;
+    } else {
+      return selectedChainBalance['keyBasedWallet'] < MAX_PLR_TOKEN_LIMIT;
+    }
+  }, [accounts, selectedAccountType, selectedFromNetwork]);
+
+  const hasEnoughPLRInPolygon = !hasNotEnoughPLRForPolygon();
```

Comment:
Coudn't this function just be `hasEnoughPlrForPolygon`? Also in this case if they haven't selected a chain balance then `hasEnoughPlrForPolygon` will be true

Also if we're checking if they have enough PLR, shouldn't we be checking `MIN_PLR_TOKEN_LIMIT` instead of the max amount?
```suggestion
  const hasEnoughPLRForPolygon = useCallback(() => {
    const selectedChainBalance = accounts.find((acc) => acc.chainId === CHAIN_ID.POLYGON);
    if (!selectedChainBalance) return false;

    if (selectedAccountType === AccountTypes.Contract) {
      return selectedChainBalance['smartWallet'] > MAX_PLR_TOKEN_LIMIT;
    } else {
      return selectedChainBalance['keyBasedWallet'] > MAX_PLR_TOKEN_LIMIT;
    }
  }, [accounts, selectedAccountType, selectedFromNetwork]);
```